sklearn/manifold/_locally_linear.py

```diff
@@ -537,7 +553,16 @@ def locally_linear_embedding(
 
             nbrs_x, nbrs_y = np.meshgrid(neighbors[i], neighbors[i])
             M[nbrs_x, nbrs_y] -= GiGiT
-            M[neighbors[i], neighbors[i]] += 1
+
+            if M_sparse:
+                for idx1 in range(len(neighbors[i])):
+                    for idx2 in range(len(neighbors[i])):
+                        M[neighbors[i][idx1], neighbors[i][idx2]] += 1
+            else:
+                M[neighbors[i], neighbors[i]] += 1
```

Comment:
The double loop is not doing what you thinking. The looping should be:

```python
for idx1 in range(len(neighbors[i])):
    M[neighbors[i][idx1], neighbors[i][idx1]] += 1
```

to have the equivalence of the dense case.

Anyway, I think we can do:

```suggestion
            M[neighbors[i], neighbors[i]] += np.ones(shape=n_neighbors)
```

That works both for dense and sparse.